packages/cxl-jwplayer/src/components/cxl-jwplayer.js

```diff
@@ -1,17 +1,152 @@
-import { LitElement, html, customElement } from 'lit-element';
+import { LitElement, html, customElement, property } from 'lit-element';
+import '@conversionxl/cxl-lumo-styles';
+// import { registerGlobalStyles } from '@conversionxl/cxl-lumo-styles/src/utils';
 import cxlJwplayerStyles from '../styles/cxl-jwplayer-css.js';
+// import cxlJwplayerGlobalStyles from '../styles/global/cxl-jwplayer-css.js';
 
 @customElement('cxl-jwplayer')
 export class CxlJwPlayer extends LitElement {
   static get styles() {
     return [cxlJwplayerStyles];
   }
 
-  static get properties() {
-    return {};
+  @property({ type: Object })
+  player = null;
+
+  @property({ type: Object })
+  config = {};
+
+  @property({ type: String })
+  mediaId = '';
+
+  constructor() {
+    super();
+    this.jwplayer = window.jwplayer || null;
+    if (!this.jwplayer) console.error('JWPlayer library not found.');
+  }
+
+  /*
+   * Implemented to avoid default, which attaches a shadowDOM tree byt default.
+   */
+  createRenderRoot() {
+    return this;
   }
 
   render() {
-    return html``;
+    return html`
+      <h2>cxl-jwplayer</h2>
+      <div id="cxl-jwplayer-container" class="cxl-jwplayer-container">
+        <div id="cxl-jwplayer" class="cxl-jwplayer jwplayer"></div>
+        ${this.config.transcript
+          ? html`
+              <div id="cxl-transcript-container" class="transcript-container">
+                <div id="searchbox" class="searchbox">
+                  <input
+                    id="search"
+                    type="search"
+                    class="search"
+                    autocomplete="off"
+                    results="5"
+                    placeholder="Type and press enter to search"
+                  />
+                  <span id="match" class="match hidden">0 of 0</span>
+                </div>
+                <div class="jw-aspect jw-reset"></div>
+                <div class="transcript-wrapper">
+                  <div id="transcript" class="transcript"></div>
+                </div>
+              </div>
+            `
+          : ''}
+      </div>
+    `;
+  }
+
+  firstUpdated() {
+    // Run player setup if minimal requirements are met
+    // Bail if jwplayer is not loaded or mediaId is not set
+    if (!this.jwplayer || !this.mediaId) return;
+
+    this.setupPlayer();
+  }
+
+  setupPlayer() {
+    /**
+     *
+     * Configure and Setup the Player
+     *
+     */
+    // Fetched via GET request to JW Platform
+    // One video is an object. Multiple videos is an array of objects.
+    // Paths are hard coded for now, while developing this implementation.
+    const data = {
+      config: {
```

Comment:
Latest media object.
Appears as variable `cxlJWPlayerData` on a page where video should be loaded. Multiple media objects are stored in this variable using the jw media id as the key. ( data = cxlJWPlayerData[jwmediaid] )

![image](https://user-images.githubusercontent.com/811103/92481697-56bb3600-f1ac-11ea-978f-bd7b3a4efd09.png)

```json
{
  "CZbUvhMT": {
    "path": "https://cxl.com/institute/wp-content/plugins/cxl-jwplayer/",
    "plugins": [],
    "media_config": {
      "title": "Online Course: Google Data Studio",
      "description": "",
      "kind": "Single Item",
      "playlist": [
        {
          "title": "Online Course: Google Data Studio",
          "mediaid": "CZbUvhMT",
          "link": "https://cdn.jwplayer.com/previews/CZbUvhMT?exp=1599571440&sig=e598ef99d1044ac616680a6eb9f9661a",
          "image": "https://cdn.jwplayer.com/v2/media/CZbUvhMT/poster.jpg?width=720",
          "images": [
            {
              "src": "https://cdn.jwplayer.com/v2/media/CZbUvhMT/poster.jpg?width=320",
              "width": 320,
              "type": "image/jpeg"
            },
            {
              "src": "https://cdn.jwplayer.com/v2/media/CZbUvhMT/poster.jpg?width=480",
              "width": 480,
              "type": "image/jpeg"
            },
            {
              "src": "https://cdn.jwplayer.com/v2/media/CZbUvhMT/poster.jpg?width=640",
              "width": 640,
              "type": "image/jpeg"
            },
            {
              "src": "https://cdn.jwplayer.com/v2/media/CZbUvhMT/poster.jpg?width=720",
              "width": 720,
              "type": "image/jpeg"
            }
          ],
          "duration": 116,
          "pubdate": 1579309116,
          "description": "",
          "tags": "course--promo,Google Data Studio,slug--google-data-studio",
          "sources": [
            {
              "default": false,
              "type": "hls",
              "file": "https://cdn.jwplayer.com/manifests/CZbUvhMT.m3u8?exp=1599571440&sig=473d4ee45af637e0f08b7417b30babf6",
              "label": "0",
              "mimeType": "application/vnd.apple.mpegurl",
              "liveSyncDuration": 25,
              "preload": "metadata"
            }
          ],
          "tracks": [
            {
              "file": "https://cdn.jwplayer.com/tracks/7yJfJuql.vtt",
              "kind": "captions",
              "label": "English"
            },
            {
              "file": "https://cdn.jwplayer.com/strips/CZbUvhMT-120.vtt",
              "kind": "thumbnails"
            }
          ],
          "variations": [],
          "asset_id": "vid-google-data-studio--course--promo",
          "wp_post_id": "1879636"
        }
      ],
      "feed_instance_id": "d9270b53-c16c-4b33-a00a-e8e0d2bcb3f9",
      "controls": true,
      "playbackRateControls": [
        0.25,
        0.5,
        0.75,
        1,
        1.25,
        1.5,
        1.75,
        2
      ],
      "displaydescription": false,
      "stretching": "uniform",
      "skin": {
        "name": "cxl-institute"
      },
      "plugins": {}
    },
    "transcript": false,
    "chapters": false
  }
}
```

Reply:
This is working nicely. It would be great to have a real multiple videos json to test.